docs/examples/viz_animated_surfaces.py

```diff
@@ -0,0 +1,202 @@
+
+
+def update_coordinates(x, y, equation, time=0, cmap_name='viridis'):
+
+    # z is the function F i.e. F(x, y, t)
+    z = eval(equation)
+    xyz = np.vstack([x, y, z]).T
+
+    # creating the colormap
+    v = np.copy(z)
+    v /= np.max(np.abs(v), axis=0)
+    colors = colormap.create_colormap(v, name=cmap_name)
+
+    return xyz, colors
+
+
+###############################################################################
+# Variables and their usage-
+# time: float
+#       initial value of the time variable i.e. value of the time variable at
+#       the beginning of the program; (default = 0)
+# dt: float
+#     amount by which the "time" variable is incremented for every iteration
+#     of timer_callback function; (default = 0.1)
+# lower_xbound: float
+#               lower bound of the x values in which the function is plotted
+#               (default = -1)
+# upper_xbound: float
+#               Upper bound of the x values in which the function is plotted
+#               (default = 1)
+# lower_ybound: float
+#               lower bound of the y values in which the function is plotted
+#               (default = -1)
+# upper_ybound: float
+#               Upper bound of the y values in which the function is plotted
+#               (default = 1)
+# npoints: int
+#          For high quality rendering, keep the number high but kindly note
+#          that higher values for npoints slows down the animation
+#          (default = 128)
+#
+
+time = 0
+dt = 0.1
+lower_xbound = -1
+upper_xbound = 1
+lower_ybound = -1
+upper_ybound = 1
+npoints = 128
+
+###############################################################################
+# creating the x, y points which will be used to fit the equation to get
+# elevation and generate the surface
+x = np.linspace(lower_xbound, upper_xbound, npoints)
+y = np.linspace(lower_ybound, upper_ybound, npoints)
+x, y = np.meshgrid(x, y)
+x = x.reshape(-1)
+y = y.reshape(-1)
+
+###############################################################################
+# Initializing wave_actors and adjusting lighting options to make the waves
+# look more aesthetically pleasant.
+
+
+def wave(x, y, equation, colormap_name):
+    xyz, colors = update_coordinates(x, y, equation=equation, time=time,
+                                     cmap_name=colormap_name)
+    wave_actor = actor.surface(xyz, colors=colors)
+    wave_actor.equation = equation
+    wave_actor.cmap_name = colormap_name
+    wave_actor.GetProperty().SetAmbient(1)
+    wave_actor.vertices = utils.vertices_from_actor(wave_actor)
+    wave_actor.no_vertices_per_point = len(wave_actor.vertices)/npoints**2
+    wave_actor.initial_vertices = wave_actor.vertices.copy() - \
+        np.repeat(xyz, wave_actor.no_vertices_per_point, axis=0)
+    return wave_actor
+
+
+###############################################################################
+# Equations to be plotted
+eq1 = "(np.sin(np.cos(3*np.pi*y)-time)*x+y*np.sin(np.cos(x*3*np.pi)-time))*\
```

Comment:
It would be nice to show the function you're rendering. You could have a dynamic label that changes when you click or hover on the captions or the surfaces of the functions.

Reply:
Can you please tell which actor/function might help with it? Also, does the said actor/function have support to display greek symbols like π?

Reply:
According to this [issue](https://gitlab.kitware.com/vtk/vtk/-/issues/17528) the best way to add text with math support is by using mathtext which basically means you put your math expression between $$, like this:

`tb.message = 'Animated 2D function:' + '$2\pi^2$'`

But first you need to make sure that mathtext is supported by getting `True` after running this: 

```python
import vtk
vtk_mathtext = vtk.vtkMathTextFreeTypeTextRenderer()
print(vtk_mathtext.MathTextIsSupported())
```

Try also adding a test for this to see if this option is enabled in all the configurations in the CI.

Reply:
Hi @guaje ! I tested it locally and the mathematical symbols are rendered when using `TextBlock2D` but don't work with the `label` actor. Should I display equations corresponding to the functions by using ``TextBlock2D``?